compute_sdk/tests/unit/test_client.py

```diff
@@ -692,12 +693,19 @@ def test_version_mismatch_only_warns_once_per_ep(mocker, gcc, mock_response, ep_
     assert mock_warn.warn.call_count == len(set(ep_ids))
 
 
-def test_client_globus_app_and_login_manager_mutually_exclusive():
-    app = mock.Mock(spec=UserApp)
-    login_manager = mock.Mock(spec=LoginManager)
-    with pytest.raises(ValueError) as excinfo:
-        gc.Client(do_version_check=False, login_manager=login_manager, app=app)
-    assert "'app' and 'login_manager' are mutually exclusive" in str(excinfo.value)
+@pytest.mark.parametrize("null_key", ("app", "authorizer", "login_manager"))
+def test_client_mutually_exclusive_auth_method(null_key: str):
+    k = {
+        "app": mock.Mock(spec=UserApp),
+        "login_manager": mock.Mock(spec=LoginManager),
+        "authorizer": mock.Mock(spec=GlobusAuthorizer),
+    }
+    k[null_key] = None
+    with pytest.raises(ValueError) as record:
+        gc.Client(do_version_check=False, **k)
+        assert "'app', 'authorizer' and 'login_manager' are mutually exclusive" in str(
+            record.value
+        )
```

Comment:
If `gc.Client(...)` raises, then the `assert` won't get executed.

Reply:
Bikeshed: The raised value is a Pytest-wrapped exception, not a record, as would be created with `pytest.warns()`.  Not a functional problem (and not a blocker), just an observation.